src/payload.rs

```diff
@@ -7,9 +8,140 @@ use core::ops::{Deref, DerefMut};
 // | SW USE                        |               ACTUAL DMA PART                                    |
 // | rssi - 1 byte | pipe - 1 byte | length - 1 byte | pid_no_ack - 1 byte | payload - 1 to 252 bytes |
 
+/// A builder for an `EsbHeader` structure
+///
+/// The builder is converted into an `EsbHeader` by calling the
+/// `check()` method.
+///
+/// ## Example
+///
+/// ```rust
+/// use esb::EsbHeaderBuilder;
+///
+/// let header_result = EsbHeaderBuilder::default()
+///     .max_payload(252)
+///     .pid(0)
+///     .pipe(0)
+///     .no_ack(true)
+///     .check();
+///
+/// assert!(header_result.is_ok());
+/// ```
+///
+/// ## Default Header Contents
+///
+/// By default, the following settings will be used:
+///
+/// | Field     | Default Value |
+/// | :---      | :---          |
+/// | pid       | 0             |
+/// | no_ack    | false         |
+/// | length    | 0             |
+/// | pipe      | 0             |
+///
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct EsbHeaderBuilder(EsbHeader);
+
+impl Default for EsbHeaderBuilder {
+    fn default() -> Self {
+        EsbHeaderBuilder(EsbHeader {
+            rssi: 0,
+            pid_no_ack: 0,
+            length: 0,
+            pipe: 0,
+        })
+    }
+}
+
+impl EsbHeaderBuilder {
+    /// Set the pipe. Must be in the range 0..=7.
+    pub fn pipe(mut self, pipe: u8) -> Self {
+        self.0.pipe = pipe;
+        self
+    }
+
+    /// Set the max payload. Must be in the range 0..=252.
+    pub fn max_payload(mut self, max_payload: u8) -> Self {
+        self.0.length = max_payload;
+        self
+    }
+
+    /// Enable/disable acknowledgment
+    pub fn no_ack(mut self, no_ack: bool) -> Self {
+        // TODO(AJM): We should probably just call this
+        // method "ack", or "enable_ack", because "no_ack"
+        // is really confusing.
```

Comment:
I agree that it's confusing, but if someone that usually uses the official sdk switches to this they could get more confused. Probably it's better to keep similar to nordic's api.

Reply:
I sort of disagree, I think that we *should* mention "no_ack" in the docs/comments, but instead use "ack", because it hurts my head thinking of double negatives and active-low settings.

But! Let's tackle that later.